### PR TITLE
Add the `curry` and `uncurry` functions

### PR DESCRIPTION
**Pull request by Dean177**
Taken from Base / Core:
https://ocaml.janestreet.com/ocaml-core/latest/doc/core_kernel/Core_kernel__/Tuple/T2/index.html
https://ocaml.janestreet.com/ocaml-core/latest/doc/core_kernel/Core_kernel__/Tuple/T3/index.html

### CODE REVIEW

**Comment by Dean177** [comment]
Sure thing

File: native/src/tablecloth.mli [hunk 1]
```diff
@@ -231,84 +231,234 @@ end
 
 module Tuple2 : sig
   val create : 'a -> 'b -> 'a * 'b
+  (** Create a 2-tuple.
+
+    Tuple2.create 3 4 == (3, 4)
+
+    let zip (xs : 'a list) (ys : 'b list) : ('a * 'b) list =
+      List.map2 Tuple2.create xs ys
```
Comment (pbiggar) [inline native/src/tablecloth.mli]: `List.map2 ~f:Tuple2.create xs ys`

File: native/src/tablecloth.mli [hunk 2]
```diff
@@ -231,84 +231,234 @@ end
 
 module Tuple2 : sig
   val create : 'a -> 'b -> 'a * 'b
+  (** Create a 2-tuple.
+
+    Tuple2.create 3 4 == (3, 4)
```
Comment (pbiggar) [inline native/src/tablecloth.mli]: Single equals here I would think.

File: native/src/tablecloth.mli [hunk 3]
```diff
@@ -231,84 +231,234 @@ end
...
+    Tuple2.mapFirst ~f:String.length ("stressed", 16) == (8, 16)
+  *)
 
   val map_first : f:('a -> 'x) -> ('a * 'b) -> ('x * 'b)
 
   val mapSecond : f:('b -> 'y) -> ('a * 'b) -> ('a * 'y)
+  (** Transform the second value in a tuple.
+
+    Tuple2.mapSecond ~f:sqrt ("stressed", 16.) == ("stressed", 4.)
+    Tuple2.mapSecond ~f:(~-) ("stressed", 16) == ("stressed", -16)
```
Comment (pbiggar) [inline native/src/tablecloth.mli]: Is this syntax valid (`~-`)? I would have though it would be `(-)`

File: native/src/tablecloth.mli (hunk 2 above)
Comment (pbiggar) [reply native/src/tablecloth.mli]: Applies to rest of docstrings too.

File: native/src/tablecloth.mli [hunk 4]
```diff
@@ -231,84 +231,234 @@ end
...
 end
 
 module Tuple3 : sig
   val create : 'a -> 'b -> 'c -> ('a * 'b * 'c)
+  (** Create a 3-tuple.
+
+    Tuple3.create 3 4 5 == (3, 4, 5)
+
+    let zip3 (xs : 'a list) (ys : 'b list) (zs : 'c list) : ('a * 'b * 'c) list =
+      List.map3 Tuple3.create xs ys zs
```
Comment (pbiggar) [inline native/src/tablecloth.mli]: Needs a `~f`

File: native/src/tablecloth.mli (hunk 3 above)
Comment (Dean177) [reply native/src/tablecloth.mli]: Yep, it's the 'negate' operator for integers: https://caml.inria.fr/pub/docs/manual-ocaml/libref/Pervasives.html#1_Integerarithmetic.

We should add an Int.negate to Tablecloth!

Comment (pbiggar) [reply native/src/tablecloth.mli]: Ah, cool. Yes we should!

**Comment by pbiggar** [comment]
This looks good to go. Can you click the rebuild button on CircleCI to run the build again?

**Comment by Dean177** [comment]
@pbiggar master has now been merged in, so CI has now been run against this PR.

**Comment by pbiggar** [comment]
(rebasing out the merges, before merging)

**Pull request merged**